lxd/images.go

```diff
@@ -99,6 +99,13 @@ func unpack(file string, path string) error {
 	} else if strings.HasPrefix(extension, ".squashfs") {
 		command = "unsquashfs"
 		args = append(args, "-f", "-d", path, "-n")
+
+		mem, err := deviceTotalMemory()
+		mem = mem / 1024 / 1024 / 10
```

Comment:
The reason is simply that unsquashfs defaults to 256MB chunks, so we only want to set the argument if the resulting computed chunk size is smaller than that and we don't want that chunk to be bigger than 10% of the system's memory.